src/PowerShellEditorServices/Workspace/WorkspaceFileSystemWrapper.cs

```diff
@@ -0,0 +1,385 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Utility;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.PowerShell.EditorServices
+{
+
+    /// <summary>
+    /// A FileSystem wrapper class which only returns files and directories that the consumer is interested in,
+    /// with a maximum recursion depth and silently ignores most file system errors. Typically this is used by the
+    /// Microsoft.Extensions.FileSystemGlobbing library.
+    /// </summary>
+    public class WorkspaceFileSystemWrapperFactory
+    {
+        private readonly DirectoryInfoBase _rootDirectory;
+        private readonly string[] _allowedExtensions;
+        private readonly bool _ignoreReparsePoints;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Gets the maximum depth of the directories that will be searched
+        /// </summary>
+        internal readonly int MaxRecursionDepth;
+
+        /// <summary>
+        /// Gets or sets the logging facility
+        /// </summary>
+        internal ILogger Logger
+        {
+            get { return _logger; }
```

Comment:
If you get rid of the `_logger` field, the compiler will create one automagically for you. Works exactly the same but fewer lines of code to wrangle

Reply:
I tried a bunch of things but I don't understand what I can get rid of here.  Can you give me a concrete example?

Reply:
Oh sorry! Yep:

```csharp
internal ILogger Logger { get; }

...

public WorkspaceFileSystemWrapperFactory(..., ILogger logger)
{
    ...
    Logger = logger;
}

...

public void DoSomething()
{
    try
    {
        DoStuff();
    }
    catch (Exception e)
    {
        Logger.WriteHandledException("Something terrible happened!", e);
    }
}
```

Reply:
Done.